### PR TITLE
Use r2dbc and sql dependencies via micronaut-r2dbc and micronaut-sql

### PR DESCRIPTION
**Pull request by radovanradic**
@dstepanov I think we talked about this, to remove these dependencies from micronaut-data and use them via micronaut-r2dbc and micronaut-sql.

### CODE REVIEW

**Comment by radovanradic** [comment]
Weird error 
```
ERROR i.m.http.server.RouteExecutor - Unexpected error occurred: No bean introspection available for type [interface io.micronaut.core.convert.value.ConvertibleValues]. Ensure the class is annotated with io.micronaut.core.annotation.Introspected
io.micronaut.core.beans.exceptions.IntrospectionException: No bean introspection available for type [interface io.micronaut.core.convert.value.ConvertibleValues]. Ensure the class is annotated with io.micronaut.core.annotation.Introspected
```

**Comment by radovanradic** [comment]
Btw, looks it is unrelated to these changes. Same failures are occurring in master. Something in core might have changed.

**Comment by radovanradic** [comment]
I see that your new PR https://github.com/micronaut-projects/micronaut-data/pull/2175 uses this 
```
micronaut = '4.0.0-M3'
micronaut-platform = '4.0.0-M2'
```
and has failures, probably the same reason. Now r2dbc is also failing with the same error

ERROR i.m.http.server.RouteExecutor - Unexpected error occurred: No bean introspection available for type [interface io.micronaut.core.convert.value.ConvertibleValues]. Ensure the class is annotated with io.micronaut.core.annotation.Introspected
io.micronaut.core.beans.exceptions.IntrospectionException: No bean introspection available for type [interface io.micronaut.core.convert.value.ConvertibleValues]. Ensure the class is annotated with io.micronaut.core.annotation.Introspected
	
Something changed in http or json recently.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,16 +1,16 @@
 [versions]
 micronaut = "4.0.0-M2"
-micronaut-platform = "4.0.0-SNAPSHOT"
+micronaut-platform = "4.0.0-M2"
```
[the remark this answers is not in the document]
Comment (dstepanov) [reply gradle/libs.versions.toml]: Core M3

**Review by dstepanov — APPROVED** [review]

**Pull request merged**